### ISSUE TITLE
System.Diagnostics.Process.BootTime doesn't work on Android

### ISSUE THREAD

**Issue by filipnavara** [dotnet/runtime#83816]
```
[DOTNET]   Error: Failure getting precise App startup time.
[DOTNET] System.UnauthorizedAccessException: Access to the path '/proc/stat' is denied.
[DOTNET]  ---> System.IO.IOException: Permission denied
[DOTNET]    --- End of inner exception stack trace ---
[DOTNET]    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
[DOTNET]    at Interop.CheckIo(Error error, String path, Boolean isDirError)
[DOTNET]    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Func`4 createOpenException)
[DOTNET]    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException)
[DOTNET]    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode, Func`4 createOpenException)
[DOTNET]    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
[DOTNET]    at System.IO.Strategies.UnixFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
[DOTNET]    at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
[DOTNET]    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
[DOTNET]    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
[DOTNET]    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
[DOTNET]    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
[DOTNET]    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
[DOTNET]    at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
[DOTNET]    at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize)
[DOTNET]    at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks)
[DOTNET]    at System.IO.File.ReadAllText(String path, Encoding encoding)
[DOTNET]    at System.IO.File.ReadAllText(String path)
[DOTNET]    at System.Diagnostics.Process.get_BootTime()
[DOTNET]    at System.Diagnostics.Process.BootTimeToDateTime(TimeSpan timespanAfterBoot)
[DOTNET]    at System.Diagnostics.Process.get_StartTimeCore()
[DOTNET]    at System.Diagnostics.Process.get_StartTime()
```

**Comment by simonrozsival** [comment]
This is a duplicate of https://github.com/dotnet/runtime/issues/67583 and it's already been fixed in .NET 8 since September. For some reason, we didn't backport it into .NET 6 and .NET 7 back then so we should backport the fix now.

**Comment by filipnavara** [comment]
> This is a duplicate of #67583 and it's already been fixed in .NET 8 since September. For some reason, we didn't backport it into .NET 6 and .NET 7 back then so we should backport the fix now.

My bad, it sounded familiar, but I missed the issue because it was marked as closed. Backport to 7.0.x would be nice.

**Comment by adamsitnik** [comment]
Fixed thanks to @am11 who provided the fix (#67589) and @simonrozsival who backported it to 6.0 and 7.0 (#83965 and #83966).
